### PR TITLE
Add example for training FSRS from CSV file

### PR DESCRIPTION
**Pull request by L-M-Sherlock**
same to https://github.com/open-spaced-repetition/fsrs-rs-nodejs/pull/12

### CODE REVIEW

File: examples/train_csv.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+
+def main():
+    # Read revlog.csv
+    # Please download from
+    # https://github.com/open-spaced-repetition/fsrs-rs/files/15046782/revlog.csv
+    with open("./revlog.csv", "r") as file:
+        records = list(csv.DictReader(file))
+
+    print(f"revlogs.len() = {len(records)}")
```
Comment (asukaminato0721) [inline examples/train_csv.py:15]:
```
print(f"{len(records) = }")
```

File: examples/train_csv.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+def group_reviews_by_card(records: List[Dict]) -> Dict[str, List[Tuple[datetime, int]]]:
+    reviews_by_card: Dict[str, List[Tuple[datetime, int]]] = {}
+
+    for record in records:
+        card_id = record["card_id"]
+        if card_id not in reviews_by_card:
+            reviews_by_card[card_id] = []
+
+        # Convert millisecond timestamp to second timestamp
+        timestamp = int(record["review_time"]) // 1000
```
Comment (asukaminato0721) [inline examples/train_csv.py:46]: https://docs.python.org/zh-cn/3.13/library/typing.html#typing.TypedDict

File: examples/train_csv.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    # Create FSRS instance and optimize
+    fsrs = FSRS([])
+    optimized_parameters = fsrs.compute_parameters(fsrs_items)
+    print("optimized parameters:", optimized_parameters)
+    end_time = time.time()
+    print(f"Full training time: {end_time - start_time:.2f}s\n")
+
+
+def group_reviews_by_card(records: List[Dict]) -> Dict[str, List[Tuple[datetime, int]]]:
```
Comment (asukaminato0721) [inline examples/train_csv.py:37]: here Dict could be https://docs.python.org/zh-cn/3.13/library/typing.html#typing.TypedDict

File: examples/train_csv.py (hunk 1 above)
Comment (L-M-Sherlock) [inline examples/train_csv.py:15]:
```suggestion
    print(f"{len(records) = }")
```

**Pull request merged**